### ISSUE TITLE
Index refresh failed (despite succeeding)

### ISSUE THREAD

**Issue by gir489returns** [TuxedoTako/4chan-xt#95]
I had been using 4chan-x for a while but decided to give XT a try today. It seems like a great update, much more maintained, and not much changed superficially. However, when I have it set to use 4Chan X's catelog by viewing the main page, it keeps giving me this error.

![image](https://github.com/user-attachments/assets/a9329a1d-992a-48ba-9ee3-94030d09db24)

However, the index DOES update.

**Comment by TuxedoTako** [comment]
What browser and user script manager are you using? Or are you using the extension version?

**Comment by gir489returns** [comment]
> What browser and user script manager are you using? Or are you using the extension version?

I am using Edge with Tampermonkey.

**Comment by TuxedoTako** [comment]
I can't reproduce the issue. Does your console show any errors?

**Comment by gir489returns** [comment]
Yes.

![image](https://github.com/user-attachments/assets/cfe69722-3b70-4357-9082-e7f6289ee0ea)

```
userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:13015  Index failure: Cannot set properties of null (setting 'disabled') TypeError: Cannot set properties of null (setting 'disabled')
    at Object.setPage (chrome-extension://iikmkjmpaadaobahmlepeloendndfphd/userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:12909:39)
    at Object.pageLoad (chrome-extension://iikmkjmpaadaobahmlepeloendndfphd/userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:12847:36)
    at Object.parse (chrome-extension://iikmkjmpaadaobahmlepeloendndfphd/userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:13045:20)
    at XMLHttpRequest.load (chrome-extension://iikmkjmpaadaobahmlepeloendndfphd/userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:13009:17)
    at XMLHttpRequest.onloadend (chrome-extension://iikmkjmpaadaobahmlepeloendndfphd/userscript.html?name=4chan-XT.user.js&id=2973cc2d-8243-4679-a9e9-f48a7854aaac:1917:19)
```

**Comment by TuxedoTako** [comment]
So either line 702 or 705.

https://github.com/TuxedoTako/4chan-xt/blob/2ec48c9333fb7d18ec247f4aa564e0f95d54f894/src/General/Index.js#L691-L705

Weird, that is referencing the pagination of the index, which is actually still on the page in catalog mode, just hidden by CSS. Do you have any other user scripts or extensions that modify the DOM?

**Comment by gir489returns** [comment]
> Weird, that is referencing the pagination of the index, which is actually still on the page in catalog mode, just hidden by CSS. Do you have any other user scripts or extensions that modify the DOM?

The only 4chan userscript I have is 4chan-captcha-solver, but it should only be activating on quick reply.

**Comment by TuxedoTako** [comment]
Can you export your settings and post them here?

**Comment by gir489returns** [comment]
[4chan XT v2.13.0-1724095292392.json](https://github.com/user-attachments/files/16737039/4chan.XT.v2.13.0-1724095292392.json)

**Comment by TuxedoTako** [comment]
Sorry, I can't reproduce it.

**Comment by gir489returns** [comment]
This seems to only occur on InPrivate when using the plugin. If I visit 4chan normally, the error does not occur.

**Comment by saxamaphone69** [comment]
Can't reproduce it either.

Does this happen on a fresh 4chan XT profile? Go to install 4chan XT again but rename it something different.

I see an element in your HTML with `div all initial` etc, your `body` has classes that aren't normally added, and your catalogue mode is missing the 'large/small' drop down option - so something else is happening. Disable everything else BUT TamperMonkey and 4chan XT.

**Comment by gir489returns** [comment]
> I see an element in your HTML with `div all initial` etc, your `body` has classes that aren't normally added, and your catalogue mode is missing the 'large/small' drop down option - so something else is happening. Disable everything else BUT TamperMonkey and 4chan XT.

![image](https://github.com/user-attachments/assets/8119aa40-3712-4c96-9f39-630a7c119481)

This is just tampermonkey with 4chan-XT in InPrivate mode.

**Comment by saxamaphone69** [comment]
The plot thickens.

When you say only in InPrivate mode, as in if you go to 4chan (with XT enabled) in a non-InPrivate mode browsing window, it works?

Does it happen only on /v/ or other boards too (like a NSFW one)? What about just the regular index (non catalogue mode)?

If you have the DevTools open again as you refresh the page, do you get anything else that stands out (either console errors or network requests)?

**Comment by gir489returns** [comment]
> When you say only in InPrivate mode, as in if you go to 4chan (with XT enabled) in a non-InPrivate mode browsing window, it works?

Yeah. Which I have to do to add MD5s or RegExs of bot shitposts to the 4chan-XT settings occasionally, otherwise they don't save.

> Does it happen only on /v/ or other boards too (like a NSFW one)? What about just the regular index (non catalogue mode)?

It happens on /pol/ too.

![image](https://github.com/user-attachments/assets/47e85c2e-e768-4220-a159-806478188e26)

> If you have the DevTools open again as you refresh the page, do you get anything else that stands out (either console errors or network requests)?

Hard-refreshing the page after I get the error makes it go away regardless. I can usually tell if the page has had the error or not, because it says ... instead of X seconds ago for the bracket that usually tells you how old the catalog refresh is. These are the only errors in the console after the hard refresh:

```
GET https://static.danbo.org/publisher/q2g345hq2g534-4chan/js/preload.4chan.js net::ERR_BLOCKED_BY_CLIENT
[Intervention] Images loaded lazily and replaced with placeholders. Load events are deferred. See https://go.microsoft.com/fwlink/?linkid=2048113
GET https://cdn.pubfuture-ad.com/v2/unit/pt.js net::ERR_BLOCKED_BY_CLIENT
```

They seem to be related to my adblocker, not 4chan-XT.

**Comment by TuxedoTako** [comment]
> first visit

It might be the native extension conflicting with 4chan XT. 4chan XT does turn of the native extension, but the first page visit it still loads.

**Comment by TuxedoTako** [comment]
Yes, I can reproduce it in private mode, and blocking the native extension with uBlock seems to fix the issue.

**Comment by gir489returns** [comment]
> Yes, I can reproduce it in private mode, and blocking the native extension with uBlock seems to fix the issue.

Hi, @TuxedoTako. Do you by chance have a RegEx or uBlock user-filter string? I'm not sure what you're referring to as the "native extension."

**Comment by TuxedoTako** [comment]
The native extension, as 4chan calls it on its settings modal, is what handles stuff like post hovering and thread updating on 4chan. 4chan X(T) takes over those functions, so running them both results in conflicts. This is an ublock origin filter you can use:

```text
||s.4cdn.org/js/extension.min.*.js$script,domain=boards.4chan.org,important
```
